integration/helpers/archive.go

```diff
@@ -0,0 +1,154 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package helpers
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"compress/gzip"
+	"context"
+	"io"
+	"log/slog"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/gravitational/teleport"
+	"github.com/gravitational/trace"
+)
+
+// CompressDirToZipFile compresses a directory into `.zip` format,
+// preserving the relative file path structure of the source directory.
```

Comment:
Seems like this contradicts the flattening comment below? https://github.com/gravitational/teleport/pull/47060/files#r1788211724

Reply:
I've changed this logic in recent commit to properly test behavior with replace in case if archive have some structure like `teleport/bin/tsh`